middleman/chat_functions.py

```diff
@@ -41,6 +41,8 @@ async def send_text_to_room(
             return "Unknown room alias"
     elif room.startswith("!"):
         room_id = room
+    elif room.startswith("@"):
+        room_id = await self.get_dm(room)
```

Comment:
This is not a class instance, as such there is no `self`. Also you need to pass in the client.

```suggestion
        room_id = await get_dm(client, room)
```